app/services/appropriate_bodies/process_batch/base.rb

```diff
@@ -32,6 +32,18 @@ def process!
         pending_induction_submission_batch.processed!
       end
 
+      # @return [Boolean] valid submissions are still valid
+      def revalidated?
+        pending_induction_submission_batch.pending_induction_submissions.without_errors.all? do |pending_induction_submission|
+          @row = pending_induction_submission_batch.row_by(trn: pending_induction_submission.trn)
+          @pending_induction_submission = pending_induction_submission
+
+          next(false) if fails_pre_checks?
+
+          validate_submission!.nil?
+        end
+      end
```

Comment:
I wonder whether revalidating submissions might be more straightforward if we were processing individual rows 💭

Is this something we have considered before?

Reply:
Not sure I'm picturing it. Here we revalidate only those rows that were previously valid, in the same (two-stage) way they got validated before. Before I merge can we chat to make sure I'm not missing a trick?